realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -248,4 +261,49 @@ static synchronized void invokeWithGlobalRefCount(RealmConfiguration configurati
         }
         callback.onResult(totalRefCount);
     }
+
+    /**
+     * Copies Realm database file from Android asset directory to the directory given in the {@link RealmConfiguration}.
+     * Copy is performed only at the first time when there is no Realm database file.
+     *
+     * @param configuration configuration object for Realm instance.
+     * @throws IOException if copying the file fails.
+     */
+    private static void copyAssetFile(RealmConfiguration configuration) {
+        File realmFile = new File(configuration.getRealmFolder(), configuration.getRealmFileName());
+        if (!realmFile.exists()) {
```

Comment:
This precondition should be checked in the constructor of the builder instead and throw an exception if it fails.